### PR TITLE
*: use cluster-aware clients, listers and informers

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Part of https://github.com/kcp-dev/kcp/issues/1943

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go [hunk 1]
```diff
@@ -111,24 +113,24 @@ func (a *Webhook) SetServiceResolver(sr webhookutil.ServiceResolver) {
...
+		return namespaceInformer.Informer().HasSynced()
 	})
 }
 
 // ValidateInitialization implements the InitializationValidator interface.
 func (a *Webhook) ValidateInitialization() error {
-	if a.hookSource == nil {
-		return fmt.Errorf("kubernetes client is not properly setup")
-	}
+	//if a.hookSource == nil {
```
Comment (ncdc) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go:131]: Do you want to keep this commented out or delete it?

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go [hunk 2]
```diff
@@ -47,10 +48,16 @@ func (r *versionDiscoveryHandler) ServeHTTP(w http.ResponseWriter, req *http.Req
...
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
 	requestedGroup := pathParts[1]
 	requestedVersion := pathParts[2]
 
-	crds, err := r.crdLister.List(req.Context(), labels.Everything())
+	crds, err := r.crdLister.Cluster(clusterName).List(req.Context(), labels.Everything())
```
Comment (ncdc) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go:60]: Doing this means we can eventually remove ctx from List() if we feel like it. That's the only reason it's there for now.

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go (hunk 1 above)
Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go:131]: We can delete, yeah

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go (hunk 2 above)
Comment (stevekuznetsov) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go:60]: Amazing! Yes, we certainly can follow-up with this.

**Review by ncdc — APPROVED** [review]

**Pull request merged**